File: Data Structures and Algorithms/LeetCode/1572. Matrix Diagonal Sum.cs
public class Solution {
    public int DiagonalSum(int[][] mat) {
        
        int sum=0;
        int j=0;
        for(int i=0;i<mat.Length;i++){
                j=i;           
                sum=sum+mat[i][i] + mat[i][mat.Length-1-i];
                if(mat.Length%2==1 & i==j & i==mat.Length/2) sum=sum- mat[i][j];
                Console.WriteLine(sum);
            
        }
        return sum;
        
    }
}

File: Data Structures and Algorithms/LeetCode/1572. Matrix Diagonal Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        total = 0
        for i in range(len(mat)):
            for j in range(len(mat[0])):
                if i == j:
                    total += mat[i][j]
                if len(mat)-i - 1 == j:
                    total += mat[i][j]
        if len(mat) % 2 != 0:
            total -= mat[len(mat)//2][len(mat)//2]
        return total
